typescript/web/src/components/spinner/index.tsx

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { Spinner as ChakraSpinner, ThemingProps } from "@chakra-ui/react";
+
+type SpinnerProps = {
+  id?: string | undefined;
```

Comment:
It's preferable to choose between `?` and `| undefined` (`?` if the property is optional, `| undefined` if you want to force the property to be set in any case).
There is no need for both (and some stricter configuration would prevent you to do both).

Reply:
Fixed